src/app/app.cpp

```diff
@@ -0,0 +1,265 @@
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <csignal>
+#include <execinfo.h>
+#include <iostream>
+
+#include <openssl/err.h>
+#include <openssl/ssl.h>
+#include <openssl/trace.h>
+
+#include <Poco/Path.h>
+#include <Poco/SignalHandler.h>
+#include <Poco/Util/HelpFormatter.h>
+#include <systemd/sd-daemon.h>
+
+#include <aos/common/version.hpp>
+#include <utils/exception.hpp>
+
+#include "app.hpp"
+#include "logger/logmodule.hpp"
+// cppcheck-suppress missingInclude
+#include "version.hpp"
+
+/***********************************************************************************************************************
+ * Static
+ **********************************************************************************************************************/
+
+static void SegmentationHandler(int sig)
+{
+    static constexpr auto cBacktraceSize = 32;
+
+    void*  array[cBacktraceSize];
+    size_t size;
+
+    LOG_ERR() << "Segmentation fault";
+
+    size = backtrace(array, cBacktraceSize);
+
+    backtrace_symbols_fd(array, size, STDERR_FILENO);
+
+    raise(sig);
+}
+
+static void RegisterSegfaultSignal()
+{
+    struct sigaction act { };
+
+    act.sa_handler = SegmentationHandler;
+    act.sa_flags   = SA_RESETHAND;
+
+    sigaction(SIGSEGV, &act, nullptr);
+}
+
+/***********************************************************************************************************************
+ * Protected
+ **********************************************************************************************************************/
+
+void App::initialize(Application& self)
+{
+    if (mStopProcessing) {
+        return;
+    }
+
+    RegisterSegfaultSignal();
+
+    auto err = mLogger.Init();
+    AOS_ERROR_CHECK_AND_THROW("can't initialize logger", err);
+
+    Application::initialize(self);
+
+    LOG_INF() << "Initialize message-proxy: version = " << AOS_MESSAGE_PROXY_VERSION;
+
+    err = mCryptoProvider.Init();
+    AOS_ERROR_CHECK_AND_THROW("can't initialize crypto provider", err);
+
+    err = mCertLoader.Init(mCryptoProvider, mPKCS11Manager);
+    AOS_ERROR_CHECK_AND_THROW("can't initialize cert loader", err);
+
+    auto retConfig = ParseConfig(mConfigFile);
+    AOS_ERROR_CHECK_AND_THROW("can't parse config", retConfig.mError);
+
+    mConfig = retConfig.mValue;
+
+    err = mIAMClient.Init(mConfig, mCertLoader, mCryptoProvider, mProvisioning);
+    AOS_ERROR_CHECK_AND_THROW("can't initialize IAM client", err);
+
+    err = mCMClient.Init(mConfig, mIAMClient, mCertLoader, mCryptoProvider, mProvisioning);
+    AOS_ERROR_CHECK_AND_THROW("can't initialize CM client", err);
+
+#ifdef VCHAN
+    mTransport.Init(mConfig.mVChan);
+#else
+    mTransport.Init(30001);
+#endif
+
+    if (mProvisioning) {
+        if (err = mCommunicationManager.Init(mConfig, mTransport); !err.IsNone()) {
+            AOS_ERROR_CHECK_AND_THROW("can't initialize communication manager", err);
```

Comment:
no need to check error for none.
`AOS_ERROR_CHECK_AND_THROW` does it

Reply:
Done